kubernetes/base/stateful-set.yaml

```diff
@@ -79,17 +64,39 @@ spec:
         args: [ "-config=/etc/consul-template/consul-template-config.hcl" ]
       # ejabberd is the core chat-service process
       - name: ejabberd
-        image: docker.dev.skillz.com:5000/ejabberd:latest
-        envFrom:
-        - secretRef:
-            name: chat-service-users
+        image: $(DOCKER_REGISTRY)/ejabberd:latest
+        imagePullPolicy: Never
+        env:
+        - name: EJABBERD_POD_PREFIX
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['app']
+        - name: EJABBERD_INTERNAL_SVC
+          value: $(SVC_NAME_INTERNAL)
+        - name: EJABBERD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: ERLANG_COOKIE
+          value: DOYOULIKEIT
+        - name: EJABBERD_K8S_AUTOCLUSTER
+          value: 'true'
+        - name: NUM_K8S_REPLICAS
+          value: '$(NUM_REPLICAS) '
```

Comment:
These values are coming from configmaps. Why not use:
```
valueFrom:
  configMapKeyRef:
    # The ConfigMap containing the value you want to assign to SPECIAL_LEVEL_KEY
    name: chat-service-config
    # Specify the key associated with the value
    key: service_name_internal
```

Reply:
I could do it that way, but since Im using the variables in more places than in `env:` definitions and container arguments, I think it's a little cleaner to use kustomize variables like this instead.

Reply:
It turns out that I was unable to use `$(NUM_REPLICAS)` the variable to set the actual number of replicas in this stateful set.
I also could not find a way to set a number of replicas in a statefulset from values in a configmap.
I've changed it back to the pattern where the `StatefulSet.chat-service.spec.replicas` is the source of truth for how many pods to run, and this `NUM_REPLICAS` variable is a reference to that value.